core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -86,16 +89,37 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
 
     import data.sparkSession.implicits._
 
-    val labels = data.map(r => r.getDouble(0) -> 1L)
-    val labelCounts = labels.groupBy(labels.columns(0)).sum(labels.columns(1)).persist()
-    val (resKeep, resDrop) = estimate(labelCounts)
+    val labels = data.map(r => r.getDouble(0))
+    val labelCounts = labels.groupBy(labels.columns(0)).count().persist()
+    val res = estimate(labelCounts)
     labelCounts.unpersist()
-    setLabels(resKeep, resDrop)
+    setLabels(res.labelsKept.toSet, res.labelsDropped.toSet, res.labelsDroppedTotal)
```

Comment:
`.distinct` perhaps instead of `.toSet` so the order would be kept?

Reply:
good idea, toSet is the current behavior